File: CPP/PowerSupply.cpp
// PowerSupply.cpp
#pragma once
#include <iostream>
#include <string>
#include "Komponen.cpp"

using namespace std;

class PowerSupply : public Komponen
{
private:
    int watt;
    string efisiensi;
public:
    PowerSupply() {}

    PowerSupply(int watt, string efisiensi, string merk, string nama) : Komponen(merk, nama)
    {
        this->watt = watt;
        this->efisiensi = efisiensi;
    }

    void setWatt(int watt)
    {
        this->watt = watt;
    }

    void setEfisiensi(string efisiensi)
    {
        this->efisiensi = efisiensi;
    }

    int getWatt()
    {
        return this->watt;
    }

    string getEfisiensi()
    {
        return this->efisiensi;
    }

    ~PowerSupply() {}
};

File: PYTHON/PowerSupply.py
class PowerSupply:
    def __init__(self, watt=0, efisiensi="", merk="", nama=""):
        self.watt = watt
        self.efisiensi = efisiensi
        self.merk = merk
        self.nama = nama

    def set_watt(self, watt):
        self.watt = watt

    def set_efisiensi(self, efisiensi):
        self.efisiensi = efisiensi

    def get_watt(self):
        return self.watt

    def get_efisiensi(self):
        return self.efisiensi

    def get_merk(self):
        return self.merk

    def get_nama(self):
        return self.nama
